state_base.py

```diff
@@ -48,6 +47,8 @@ def get_state(self, group):
         initial state configured for the block. It will also set that as the
         state.
         """
+        # initial state is set once a signal passes through
+        self._initial_state = self.initial_state(Signal())
         if group not in self._states:
             self._states[group] = copy(self._initial_state)
```

Comment:
I can't recall what this `copy` is here for but I'm thinking it's not necessary anymore based on the new code. We probably don't even need the `self._initial_state` attribute